File: swift/robot-name/Sources/RobotName/RobotName.swift
class Robot {
    
    private static let ALPHABET: [String] = [
        "A", "B", "C", "D", "E", "F", "G", "H", "I", "J", "K", "L", "M",
        "N", "O", "P", "Q", "R", "S", "T", "U", "V", "W", "X", "Y", "Z"
    ]
    private static var usedNames: Set<String> = []
    var name: String
    
    init() {
        self.name = Robot.generateName()
    }
    
    func resetName() {
        Robot.usedNames.remove(self.name)
        self.name = ""
    }
    
    private static func generateName() -> String {
        let name = "\(randomLetter())\(randomLetter())\(randomNumber())\(randomNumber())\(randomNumber())"
        
        usedNames.insert(name)
        
        return name
    }
    
    private static func randomLetter() -> String {
        return ALPHABET[Int.random(in: 0 ..< 26)]
    }
    
    private static func randomNumber() -> Int {
        return Int.random(in: 0 ... 9)
    }
}

File: python/robot-name/robot_name.py
import random
import string


class Robot:

	used_names = set()

	def __init__(self):
		self.name = self.generate_name()

	def generate_name(self) -> str:
		name = ""

		for i in range(5):
			name += random.choice(string.ascii_uppercase if i < 2 else string.digits)

		if name in self.used_names:
			name = self.generate_name()

		self.used_names.add(name)

		return name

	def reset(self):
		self.name = self.generate_name()